### DISCUSSION TITLE
Session Memory Team and all AssistanrAgent

### DISCUSSION THREAD

**Discussion by asheeshgarg** [microsoft/autogen#4328] [General]
@ekzhu  do we have a example of AssitantChatAgent and Team communication with some sharing of session memory between all agents or passing the messsage context or custom object.

**Comment by ekzhu** [comment]
The group chat team already share context. For decoupling of agent and memory we are addressing this with memory module. So you can share and load modules between different agents.

**Comment by asheeshgarg** [comment]
Just like in the core reflection notebook where we have a session establish between code reviewer and code generator where a state is passed between two agents. I don't see a way to pass either message context, or any custom object in AssitantAgentChat between agent apart from the Chat message. Do we have any branch on Memory work?

**Reply by ekzhu** [reply]
The AgentChat group chat teams: RoundRobinGroupChat, SelectorGroupChat, and Swarm, all share context by broadcasting the chat messages from speaker to every other agents. So they already share context. 

There is currently no way to pass custom objects through chat messages in AgentChat. For this we need users (like you) to provide us some concrete usage cases to help us design. 

For memory, the tracking issue is https://github.com/microsoft/autogen/issues/4039.

**Reply by asheeshgarg** [reply]
thanks @ekzhu is memory targeted for 0.4 milestone

**Reply by ekzhu** [reply]
on track
